{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "*JSC270, Winter 2020 - Prof. Taback*\n",
    "\n",
    "# <center>  Laboratory 7.1 - \"Survival Analysis\"  </center>\n",
    "## <center> Lab assignment </center>\n",
    "\n",
    "### <center> Date: March 4, 12:00 - 14:00, BA3175 <br><br> Due Date: March 9, 12:00pm (noon) \n",
    "<center>  This assignment is a group assignment. <br>You may work with the teammate of your choice.</center>\n",
    "\n",
    "\n",
    "***"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<div style=\"background-color: #f7f7f7; padding:10px;\">\n",
    "    \n",
    "<h3> Objectives </h3>\n",
    "\n",
    "<ul>\n",
    "    <li> Understand and apply survival analysis </li>\n",
    "    <li> Understand supervised versus unsupervised learning </li>\n",
    "    <li> Gain experience applying hierarchical clustering </li>\n",
    "    <li> Gain experience working with Cancer Genomic data </li>\n",
    "</ul>\n",
    "\n",
    "<h3> Instructions </h3>\n",
    "\n",
    "<ul>\n",
    "    <li> Make a copy of this notebook: <a href=\"https://classroom.github.com/g/Cp1W6hU4\">https://classroom.github.com/g/Cp1W6hU4</a> </li>\n",
    "    <li> Answer the questions in the cells that indicate where your answers should be placed.  </li>\n",
    "     <li> Make sure that you explain your solutions when asked and comment your code for readability. </li>\n",
    "    <li> Commit and push changes to the Github classroom repository that has been provided (more details below).</li>\n",
    "    <li> Submit a pdf report to Quercus (more details below).</li>\n",
    " </ul>\n",
    "    \n",
    "\n",
    "\n",
    "</div>\n",
    "\n",
    "***\n",
    "\n",
    "<br>\n",
    "\n",
    "\n",
    "Can you identify distinct groups (biological “subtypes”) of patients with pancreas cancer, using The Cancer Genome Atlas (TCGA) transcriptome sequencing data, that have different survival times?  \n",
    "\n",
    "### Specific Questions\n",
    "\n",
    "**Question 1.** Merge the clinical and RNA data.\n",
    "\n",
    "**Question 2.** Generate overall Kaplan-Meier plot.\n",
    "\n",
    "**Question 3.** Use hierarchical clustering to find 2 and 3 subgroups.\n",
    "\n",
    "**Question 4.** Test to see if the there is a statistical difference in survival between the subgroups you found in Question 3.\n",
    "\n",
    "**Question 5.** Interpret the results in Question 4.\n",
    "\n",
    "<hr>\n",
    "\n",
    "\n",
    "## Data\n",
    "\n",
    "\n",
    "The data was obtained from the [Genomic Data Commons Data Portal](https://portal.gdc.cancer.gov), by issuing the following query. \n",
    "\n",
    "![](gdc_query.png)\n",
    "\n",
    "### Data Definitions\n",
    "\n",
    "The data come from different studies conducted at different institutions and are submitted to the Genomic Data Commons (GDC). The data requirements are outlined [here](https://gdc.cancer.gov/content/selecting-common-cross-study-clinical-data-elements#dmwg), and a data dictionary can be found [here](https://docs.gdc.cancer.gov/Data_Dictionary/viewer/).  \n",
    "\n",
    "\n",
    "### Linking Clinical Data and Genetic Expression Data\n",
    "\n",
    "- *Clinical data* is in the file `clinical.tsv` in the folder beginning with the name `clinical.cart`. \n",
    "\n",
    "- *Gene expression data* is in the folder beginning with `gdc_download`.  There are approximately 60000 genes measured on each patient.  An explanation of the measurement is [here](https://docs.gdc.cancer.gov/Encyclopedia/pages/HTSeq-FPKM/)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    " <br>\n",
    " <br>\n",
    " \n",
    " # <p style=\"background-color: #EEEEEE;text-align:center;padding-top: 15px;padding-bottom: 15px;\" >Submission and Grading </p>\n",
    " \n",
    "## Submission\n",
    "\n",
    "**What should be submitted?**\n",
    "\n",
    "* A Jupyter notebook using Python that contains all the data analyses. The flow of the notebook should be Markdown and Code cells weaved together. In addition make sure to comment your code in the code cells.\n",
    "\n",
    "* Additional files that your notebook requires to compile properly (e.g. a data file).\n",
    "\n",
    "* A report from your Jupyter notebook exported as a html file without the code cells. This can be done using [nbconvert](https://nbconvert.readthedocs.io/en/latest/).\n",
    "\n",
    "**When should it be submitted?**\n",
    "* At the end of the lab session (in-lab progress): Submit your current progress, i.e., <mark><b>before 2:30pm on the day of the laboratory, March 4.</b></mark> (Submit to Github Classroom, and Quercus)\n",
    "* Homework (lab assignment completion): Upload your completed lab assignment by the next class, on <mark><b>Monday March 9 at 12:00pm (noon)</b></mark>. (Submit to Github Classroom, and Quercus)\n",
    "\n",
    "**Where should the assignment be submitted?**\n",
    "* Commit your finished notebook (ipynb and html versions), and any other files on the Github Classroom repository.\n",
    "* Submit a pdf version of this notebook, including code cells (print this notebook to pdf) to Quercus.\n",
    "\n",
    "\n",
    "## Grading\n",
    "\n",
    "This laboratory assignment will be graded according to the following marking scheme.\n",
    "\n",
    "| | Marks|\n",
    "|--- |--- |\n",
    "| In-lab progress | 5 |\n",
    "| Lab assignment completion | 10 |\n",
    "| Attendance for entire lab | 5 |\n",
    "| **Total** | **20**|\n",
    "\n",
    "\n",
    "Note that marks will be deduced for the following reasons: notebook doesn't compile; files are missingl; instructions are not followed.\n",
    "\n",
    "\n",
    "### In-lab progress grading rubric\n",
    "\n",
    "In-lab progress rubric is as follows (also available [here](https://jsc270.github.io/tut_grading.html#in-lab_progress)):\n",
    "\n",
    "|Category|5|4|3|2|1|\n",
    "|---|:---:|:---:|:---:|:---:|:---:|\n",
    "|**Progress (5 pts)** | Excellent progress. Optimal use of lab time. The team was fully focused on getting the assignment done, and fully engaged. Excellent progress. | Good progress. Good use of lab time. The team was focused on getting the assignment done, and engaged. | Moderate progress. Used some of the lab time well. There was some focus on getting the assignment done, but the team was occasionnally distracted. | Insufficient progress. Poor use of lab time. The team was generally distracted, and did not focus much on the assignment. | Minimal progress. The team did barely use lab time to make progress on the assignment, and was often distracted. |\n",
    "\n",
    "<br>\n",
    "\n",
    "### Lab assignment completion grading rubric\n",
    "\n",
    "Lab assignment completion grading rubric is as follows (also available [here](https://jsc270.github.io/tut_grading.html#lab_assignment_completion)):\n",
    "\n",
    "|Category|5|4|3|2|1|\n",
    "|---|:---:|:---:|:---:|:---:|:---:|\n",
    "|**Planning of code and statistical analysis <br> (2.5 pts)** | Introductory comment describes overall strategy and gives evidence of preliminary planning. Thoughtful problem decomposition breaks the problem into independent pieces that can be solved easily. | Evidence of planning before coding, but some flaws in overall strategy. | More planning needed: overall strategy ok, but have missed some obvious ways of making the code simpler. | It all hangs together, but planning was absent or rushed. | No evidence of planning. Strategy deeply flawed.|\n",
    "|**Programming execution and reproducibility of results <br> (2.5 pts)**| Mastery of Python vocabulary means that the absolute minimum amount of code is used to get the job done. Code free from duplication. Each function encapsulates a single task, and repeated tasks are performed by functions, not copy and paste. | Workable, but not elegant.  Common programming idioms used to reduce code. | The code works, but copy-paste used often. | Most of the code works, but some parts do not work. Copy-paste used very often. | Functions used inapproriately, or existing functions reinvented. Extensive use of copy and paste. |\n",
    "|**Programming Clarity <br> (2.5 pts)** | Code is a pleasure to read, and easy to understand. Code and comments form part of a seamless whole. | Comments used to discuss the why, and not how of code; to provide insight into complicated algorithms; and to indicate purpose of function (if not obvious from its name). Comment headings used to separate important sections of the code. | Generally easy to read, but some comments used inappropriately: either too many, or too few. Some variable names confusing. | Hard to understand. Poor choice of names and comments do not generally aid understanding. | Cannot understand code. The reader cannot understand why the code works. |\n",
    "|**Communication <br>(2.5 pts)**| Findings very well organised. Clear headings demarcate separate sections. Excellent flow from one section to the next. The report is easy to scan.  Tables and graphics carefully tuned and placed for desired purpose. | Findings well organised and sections clearly separated, but flow is lacking. Each section has clear purpose.  Tables and graphics clear and well chosen. | Generally well organised, but some sections muddled.  Tables or graphics appropriate, but some are poorly presented - e.g. too many decimal places, poorly chosen aspect ratio etc. | Sections unclear and no attempt to flow from one topic to the next. Graphics and tables poorly chosen to support questions. Some have fundamental flaws. | It is hard to read your notebook. There are no headings, figures are far away from where they are referenced in the text. There is no summary or conclusion. |"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
